src/common/copilot/user-feedback/CESSurvey.ts

```diff
@@ -0,0 +1,128 @@
+
+  const feedbackData = {
+    IsDismissed: false,
+    ProductContext: [
+      {
+        key: 'sessionId',
+        value: sessionId
+      },
+      {
+        key: 'scenario',
+        value: 'ProdDevCopilot'
+      }
+    ],
+    Feedbacks: [
+      {
+        key: 'comment',
+        value: ''
+      }
+    ]
+  };
+
+  const apiToken: string = await npsAuthentication(SurveyConstants.AUTHORIZATION_ENDPOINT)
+
+  const endpointUrl = `https://world.tip1.ces.microsoftcloud.com/api/v1/portalsdesigner/Surveys/powerpageschatgpt/Feedbacks?userId=${userID}`;
+  // Handle messages from the webview
+
+  feedbackPanel.webview.onDidReceiveMessage(
+    async message => {
+      switch (message.command) {
+        case 'feedback':
+          feedbackData.Feedbacks[0].value = message.text;
+          try {
+            const response = await fetch(endpointUrl, {
+              method: 'POST',
+              headers: {
+                'Content-Type': 'application/json',
+                Authorization: 'Bearer ' + apiToken,
+              },
+              body: JSON.stringify(feedbackData)
+            });
+
+            if (response.ok) {
+              // Feedback sent successfully
+              const responseJson = await response.json();
+              const feedbackId = responseJson.FeedbackId;
+              sendTelemetryEvent(telemetry, { eventName: UserFeedbackSuccessEvent, FeedbackId: feedbackId });
+            } else {
+              // Error sending feedback
+              sendTelemetryEvent(telemetry, { eventName: UserFeedbackFailureEvent, error: response.statusText });
+            }
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          } catch (error: any) {
+            // Network error or other exception
+            sendTelemetryEvent(telemetry, { eventName: UserFeedbackFailureEvent, exception: error.message });
+          }
+
+          feedbackPanel.dispose();
+      }
+      return;
+
+    },
+    undefined,
+    context.subscriptions
+  );
+
+}
+
+function getWebviewContent(feedbackCssUri: vscode.Uri, feedbackJsUri: vscode.Uri, nonce: string) {
+
+  return `<!DOCTYPE html>
+    <html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <link href="${feedbackCssUri}" rel="stylesheet">
+        </link>
+        <title>Cat Coding</title>
+    </head>
+    <body>
+    <form id="feedbackForm">
+    <label for="feedbackText" class="form-label" id="form-label"> Tell us more.</label>
+    <br/>
+    <textarea id="feedbackText" name="feedbackText" rows="5" required></textarea>
+    <br/>
+    <p class="privacy-statement">Try and be as specific as possible. Your feedback will be used to improve Copilot. <a href="https://privacy.microsoft.com/en-US/data-privacy-notice"> View privacy details </a> </p>
+    <button type="submit" class="submit-feedback">Submit</button>
```

Comment:
They don't need localization?

Reply:
Currently webview doesn't support localization. There is a workaround to do that but will pick that later.